src/Common/Integrations/Plugin_Merge_Provider_Abstract.php

```diff
@@ -0,0 +1,326 @@
+<?php
+/**
+ * Abstract for Integrations.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Integrations
+ */
+
+namespace TEC\Common\Integrations;
+
+use TEC\Common\Contracts\Service_Provider;
+use Tribe__Admin__Notices;
+use Tribe__Settings_Manager;
+
+/**
+ * Class Plugin_Merge_Provider_Abstract
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Integrations
+ */
+abstract class Plugin_Merge_Provider_Abstract extends Service_Provider {
+	/**
+	 * If the plugin was updated from a version that is less than the merged plugin version.
+	 *
+	 * @var bool
+	 */
+	protected bool $updated_to_merge_version = false;
+
+	/**
+	 * Get the plugins version where the merge was applied.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	abstract public function get_merged_version(): string;
+
+	/**
+	 * Get version key for the last version option.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	abstract public function get_last_version_option_key(): string;
+
+	/**
+	 * Get the key of the plugin file, e.g. path/file.php.
+	 *
+	 * @return string
+	 */
+	abstract public function get_plugin_file_key(): string;
+
+	/**
+	 * Get the slug of the notice to display with various notices.
+	 *
+	 * @return string
+	 */
+	abstract public function get_merge_notice_slug(): string;
+
+	/**
+	 * Get the message to display when the parent plugin is being updated but the child plugin is not active.
+	 *
+	 * @return string
+	 */
+	abstract public function get_updated_notice_message(): string;
+
+	/**
+	 * Get the message to display when the parent plugin is being updated to the merge.
+	 *
+	 * @return string
+	 */
+	abstract public function get_updated_merge_notice_message(): string;
+
+	/**
+	 * Get the message to display when the child plugin is being activated.
+	 *
+	 * @return string
+	 */
+	abstract public function get_activating_merge_notice_message(): string;
+
+	/**
+	 * Binds and sets up implementations.
+	 *
+	 * @since TBD
+	 */
+	public function register() {
+		// Was updated from a version that is less than the merged version?
+		$this->updated_to_merge_version = $this->did_update_to_merge_version();
```

Comment:
Maybe worth putting in a public getter (that sets the value if unset)?

Then we don't have to set it here and when we check it later it will be set as needed.

Reply:
I thought about making it lazy, but it needs to be set early because the setting changes in the main provider on plugin initialization via the updater or some other means, so I wanted to force it here ASAP. 

Thought about throwing it in the constructor too. Perhaps that would be better..